File: types/receipt.ts
import { Evidence } from "./evidence";

export interface ReceiptEvidenceItem {
  source: string;
  detail: string;
}

export interface ComparisonRow {
  label: string;
  value: string;
}

export interface Receipt {
  id?: string;
  decision_id: string;
  evidence_ids: string[];
  confidence: number; // 0 to 1
  confidencePct?: number; // 0 to 100
  reasoning: string;
  created_at?: string;

  // Additional detail fields for panels
  evidence?: Evidence[];
  evidenceSources?: ReceiptEvidenceItem[];
  comparisonRows?: ComparisonRow[];
  recommendation?: string;
  estimatedImpact?: string;
}

File: models/receipt.py
# Receipt model definition matching 03-DATABASE-SCHEMA.md

class Receipt:
    def __init__(self, id: str, decision_id: str, confidence: float, reasoning: str, evidence_ids: list):
        self.id = id
        self.decision_id = decision_id
        self.confidence = confidence
        self.reasoning = reasoning
        self.evidence_ids = evidence_ids
